source/Engine.cpp

```diff
@@ -1,9 +1,12 @@
 #include "Engine.h"
+#include "Subsystem/Yellowstone/Yellowstone.h"
 
 #include "Utility/Assert.h"
 
 Engine::Engine()
+	: myYellowstone(nullptr)
```

Comment:
I think it would be nice to rely on forward declarations. It depends on our design for the Engine class, because it would be a singleton that can be accessed in any game/project then it would be weird to include all the code in the header I think.